lib/ipam/ipam_test.go

```diff
@@ -1133,7 +1186,7 @@ var _ = testutils.E2eDatastoreDescribe("IPAM tests", testutils.DatastoreEtcdV3,
 		// - Assign should not return no error.
 		// - ReleaseIPs should return a slice with one (unallocatedIPs) and no error.
 		Entry("Assign 1 IPv4 address with AssignIP then try to release 2 IPs (assign one and release it)", net.IP{}, true, []string{"192.168.1.0/24", "fd80:24e2:f998:72d6::/120"}, net.ParseIP("192.168.1.0"), 0, []cnet.IP{}, nil),
-		Entry("Assign 1 IPv4 address with AssignIP then try to release 2 IPs (release a second one)", net.ParseIP("192.168.1.1"), false, []string{"192.168.1.0/24", "fd80:24e2:f998:72d6::/120"}, net.IP{}, 0, []cnet.IP{{net.ParseIP("192.168.1.1")}}, nil),
+		Entry("Assign 1 IPv4 address with AssignIP then try to release 2 IPs (release a second one)", net.ParseIP("192.168.1.1"), false, []string{"192.168.1.0/24", "fd80:24e2:f998:72d6::/120"}, net.IP{}, 0, []cnet.IP{cnet.MustParseIP("192.168.1.1")}, nil),
```

Comment:
Getting failure:

```

  Expected

      <[]net.IP | len:1, cap:1>: [

          {IP: [192, 168, 1, 1]},

      ]

  to equal

      <[]net.IP | len:1, cap:1>: [

          {

              IP: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 255, 255, 192, 168, 1, 1],

          },

      ]



  /go/src/github.com/projectcalico/libcalico-go/lib/ipam/ipam_test.go:1157
```

This is presumably because gomega `Equal` matcher is using `reflect.DeepEqual` rather than `net.IP.Equal`. Now that the de-duping going from IP -> String -> IP the embedded IPv4/v6 struct is getting mixed up in the tests :(